### ISSUE TITLE
Save list

### ISSUE THREAD

**Issue by Reginer** [Raizlabs/DBFlow#996]
DBFlow Version: 3.1.1
Question

Description:

Hey guys。

I want  to save a list like   3.0.0beta5

new SaveModelTransaction<>(ProcessModelInfo.withModels(users)).onExecute();

how I should do in 3.1.1  ?

Have a simple code  not like   

FlowManager.getDatabase(MyDatabase.class)
                    .beginTransactionAsync(new ProcessModelTransaction.Builder<>(
                            new ProcessModelTransaction.ProcessModel<FindContent.FindContentBean>() {
                                @Override
                                public void processModel(FindContent.FindContentBean content) {
                                    // do work here -- i.e. user.delete() or user.update()
                                    content.save();
                                }
                            }).addAll(event.getData()).build())  // add elements (can also handle multiple)
                    .error(new Transaction.Error() {
                        @Override
                        public void onError(Transaction transaction, Throwable error) {

                        }
                    })
                    .success(new Transaction.Success() {
                        @Override
                        public void onSuccess(Transaction transaction) {

                        }
                    }).build().execute();

Too cumbersome

**Comment by trevjonez** [comment]
You can do it synchronously.
```java
FlowManager.getDatabase(MyDatabase.class)
           .executeTransaction(databaseWrapper -> {
               for (MyModel model : models) {
                   model.save();
               }
           });
```

or consider using the save queue thread if it isn't critical that they are saved at exactly that moment.

```java
FlowManager.getDatabase(MyDatabase.class)
           .getTransactionManager()
           .getSaveQueue()
           .addAll(models);
```

Keep in mind that you can customize the save queue or explicitly flush it after calling addAll

**Comment by Reginer** [comment]
In

FlowManager.getDatabase(MyDatabase.class)
           .getTransactionManager()
           .getSaveQueue()
           .addAll(models);
Error is

addAll
(java.util.Collection<com.raizlabs.android.dbflow.structure.Model>)
in DBBatchSaveQueue cannot be applied
to
(java.util.List<FindContentBean>)

The method  addAll  can  not be  ArrayList？

**Comment by trevjonez** [comment]
You are correct the method in question has an issue with type signature. 
It should be:
`addAll(final Collection<? extends Model> list)` 
but it is: 
`addAll(final Collection<Model> list)`

**Comment by trevjonez** [comment]
As a workaround for now you can wrap the array list.
so the current `.addAll(models)` would become `.addAll(new ArrayList<>(models))`
One of us should put in a PR that fixes it, although I believe this would be a binary compatibility breaking change which depending on how @agrosner feels may not be released too soon.

You could also for the repo, make the change and request the dependencies though jitpack using your own username and commit hash as the version.

**Comment by Reginer** [comment]
I see，thanks.

**Comment by agrosner** [comment]
Itll be fixed for when 4.0.0-beta1 is released. in `develop` now.
